docs/source/frameworks/onnx.rst

```diff
@@ -0,0 +1,329 @@
+
+which will result:
+
+.. code-block:: bash
+
+   Model(tag="super_resolution:lwqr7ah5ocv3rea3", path="~/bentoml/models/super_resolution/lwqr7ah5ocv3rea3/")
+
+.. note::
+
+   ``save_model`` will use ``{"run": {"batchable": False}}`` as
+   default signatures if ``signatures`` is not provided. Set
+   ``batchable`` to ``False`` will disable BentoML's
+   :ref:`guides/batching:Adaptive Batching` functionality. That's why
+   we provide our own signatures here. Read more about :ref:`Model
+   Signatures <concepts/model:Model Signatures>` and :ref:`Batch Input
+   <concepts/model:Batching>`
+
+.. seealso::
+
+   ``save_model`` also has some :ref:`general options
+   <concepts/model:Save A Trained Model>` for functionalities like
+   saving metadata and custom objects.
+
+
+Loading an ONNX model with BentoML
+----------------------------------
+
+We can use ``load_model`` to load an ONNX model back to memory:
+
+.. code-block:: python
+
+   ort_session = bentoml.onnx.load_model("onnx_super_resolution")
+
+.. note::
+
+   BentoML will load an ONNX model back as an
+   ``onnxruntime.InferenceSession`` object which is ready to do
+   inference
+
+
+Then we can do the inference:
+
+.. code-block:: python
+
+   test_input = np.random.randn(2, 1, 244, 244) # can accept arbitrary batch size
+   ort_session.run(None, {"input": test_input.astype(np.float32)})
+
+.. note::
+
+   In above codes we need explicitly to convert input ndarray to
+   float32 because ``onnxruntime.InferenceSession`` only expects
+   single floats. In the following section we will see that BentoML
+   runner will automatically cast input data to this type
+
+
+In addition to :ref:`general options <concepts/model:Retrieve a saved
+model>` provided by :code:`load_model`, you can pass ``providers`` to
+set ONNX Runtime's `Execution Providers
+<https://onnxruntime.ai/docs/execution-providers/>`_. You can also
+pass ``session_options`` to set ``onnxruntime.InferenceSession``'s
+`SessionOptions
+<https://onnxruntime.ai/docs/api/python/api_summary.html#sessionoptions>`_.
+
+.. code-block:: python
+
+   ort_session = bentoml.onnx.load_model(
+       "super_resolution",
+       providers=["TensorrtExecutionProvider", "CUDAExecutionProvider", "CPUExecutionProvider"]
+   )
+
+
+Using Runners
+-------------
+
+.. seealso::
+
+   :ref:`Runners<concepts/runner:Using Runners>` for more information on what is
+   a Runner and how to use it.
+
+.. seealso::
+
+   :ref:`Runners<concepts/runner:Specifying Required Resources>` on how to
+   provide options for a runner.
+
+
+To use ``onnx`` runner locally, access the model via ``get`` and
+convert it to a runner:
+
+.. code-block:: python
+
+   test_input = np.random.randn(2, 1, 244, 244)
+
+   runner = bentoml.onnx.get("super_resolution").to_runner()
+
+   runner.init_local()
+
+   runner.run.run(test_input)
+
+.. note::
+
+   You don't need to cast your input ndarray to ``np.float32`` for
+   runner input
```

Comment:
I think we can remove this part in light of a Service example with ONNX.

Reply:
Also a good note about the `runner.run.run` behaviour and why it named that way.